lib/flexirest/request.rb

```diff
@@ -543,7 +543,7 @@ def do_request(etag)
           else
             _, @base_url, @url = parts
           end
-          if using_basic_auth? && model_class.basic_auth_method != :header
+          if using_basic_auth? && model_class.basic_auth_method = :url
```

Comment:
Am I missing something? You're using an assignment rather than a conditional check

Reply:
Woops, I think it went unnoticed because this path is only tested by `"should allow requests directly to URLs"` : 

https://github.com/flexirest/flexirest/blob/98c9430ce78aed57f54a5a78f9ac050c61f4e79a/spec/lib/request_spec.rb#L1273

This methods begs for a refactor, I might try that in a separate PR if I have time.